### ISSUE TITLE
Single-quoted heredoc string is mislabeled with ASCII-8BIT encoding

### ISSUE THREAD

**Issue by mojavelinux** [oracle/truffleruby#1563]
TruffleRuby incorrectly assigns the ASCII-8BIT encoding to a single-quoted heredoc string (in cases when we expect the encoding to be UTF-8). (This happens regardless of whether the string contains a multi-byte character, though is most problematic when it does).

Here's an example:

```rb
str = <<'EOS'
日
EOS
str.encoding
#=> ASCII-8BIT
```

The problem goes away if the single quotes are removed:

```rb
str = <<EOS
日
EOS
string.encoding
#=> UTF-8
```

This happens for all heredoc variations (`<<`, `<<-`, and `<<~`).

**Comment by nirvdrum** [comment]
Thanks for the easy reproduction! I'll take a look and see what's going on here.

**Comment by nirvdrum** [comment]
This has been fixed via d3229f3dd3228f36d5e095d6467bcd932f698697. I wasn't able to get it before the 1.0.0 RC 12 release, so we'll have to wait for the next release for the fix to appear in a build. I'm going to close the issue out now, but we can reopen it after the next release if necessary.

**Comment by mojavelinux** [comment]
Thanks @nirvdrum! That's great news.
